src/Caliburn.Micro.Core/ConductorBaseWithActiveItem.cs

```diff
@@ -39,12 +39,12 @@ object IHaveActiveItem.ActiveItem
         /// <returns>A task that represents the asynchronous operation.</returns>
         protected virtual async Task ChangeActiveItemAsync(T newItem, bool closePrevious, CancellationToken cancellationToken)
         {
-            await ScreenExtensions.TryDeactivateAsync(_activeItem, closePrevious, cancellationToken);
-            newItem = EnsureItem(newItem);
-
             _activeItem = newItem;
             NotifyOfPropertyChange(nameof(ActiveItem));
 
+            await ScreenExtensions.TryDeactivateAsync(_activeItem, closePrevious, cancellationToken);
+            newItem = EnsureItem(newItem);
```

Comment:
think you need to move this above _activeItem = newItem;

Reply:
You're right, I overlooked this call. Commited [10180d1](https://github.com/Caliburn-Micro/Caliburn.Micro/pull/907/commits/10180d1d3b1d8a71716ecd919debe60b4eb957ea)

Reply:
Are you still having issues with INotifyPropertyChanged?

Reply:
Do you mean the `NotifyOfPropertyChange(nameof(ActiveItem));` call?

I meant that this call should have prevented the bug on the unfixed master branch because it should trigger a UI refresh, but it have not and I don't know why. Please check my [bug-async-calls-in-ondeactivateasync](https://github.com/darxis/Caliburn.Micro/tree/bug-async-calls-in-ondeactivateasync) branch where I modified the `Caliburn.Micro.Tutorial.Wpf` example to reproduce this bug.

Reply:
> I think it what happens is the following:
> 1. When you click the 'Some other view' tab the Dispatcher (or some other thread) sets the `ActiveItem` with the `SomeOtherViewModel` instance. 
> 2. The `ActiveItem` setter call returns before the `OnDeactivateAsync` method is fully executed as the `ActivateItemAsync` call is not awaited so the `_activeItem` is not changed because it is updated after `OnDeactivateAsync` is completed. That means that the `ActiveItem` setter call returns but the `ActiveItem` would return the old viewmodel until the `OnDeactivateAsync` completes and `_activeItem` is changed.
> 3. The thread/code that set the `ActiveItem` marks that property as "Dirty" and triggers refreshing of that property before the underneath `_activeItem` field is changed - you can check that by putting a breakpoint on the `ActiveItem` setter.

But then the `OnDeactivateItem` finally completes asynchronously and the `_activeItem` is set and the `NotifyOfPropertyChange(nameof(ActiveItem));` should work, but it is not working and the UI still displays the old view.

Reply:
Ok I will look

Reply:
It seems like whatever the thing that should handle the `PropertyChanged(ActiveItem)` event and refresh the UI is just ignoring this event.

Reply:
> Okay, I managed to find out why the `TabControl` doesn't update the UI despite `PropertyChanged(ActiveItem)` being called. It is because when you click the second tab on the UI to show it then it is the `TabControl` that sets the `ActiveItem` and internally `TabControl` will remember this newly selected tab as the active tab. And later when the `PropertyChanged(ActiveItem)` is called the `TabControl` compares if the active tab it remembered is different than the new one because if that's the case there is no need to do anything. In this case it is the same tab that's why the UI doesn't update to the newly selected tab view.
> 
> So `TabControl` has the correct tab remembered but it displays the view for the old tab as `ActiveItem/_activeItem` is not updated yet because it is updated asynchronously later.
> 
> Also this bug doesn't occur if you activate the other tab programatically by calling `ActiveItem = _someOtherViewModel` from within the `Conductor`, because TabControl doesn't know the new tab yet, it get to know it only when the `NotifyOfPropertyChange(nameof(ActiveItem))` is called and then it refreshes the UI with the correct view.
> 
> TL;DR - everything is fine with the `NotifyOfPropertyChange(nameof(ActiveItem));` call and `ActiveItem` should set its backed field `_activeItem` synchronously before exiting the setter, not asynchronously within a new `Task` - the PR I submitted fixes this behavior.